knimin/lib/data_access_pm.py

```diff
@@ -0,0 +1,904 @@
+           table must have exactly the three columns:
+           [tbl_name_id, name, notes]
+    """
+    with TRN:
+        # check that tbl_name exists in DB
+        sql = """SELECT DISTINCT table_name from INFORMATION_SCHEMA.COLUMNS
+                 WHERE table_schema = 'pm' AND table_name = %s"""
+        TRN.add(sql, [tbl_name])
+        if len(TRN.execute_fetchindex()) != 1:
+            raise LabadminDBError('Table %s does not exist in data base.' %
+                                  tbl_name)
+
+        # check that table has exactly three columns
+        sql = """SELECT column_name from INFORMATION_SCHEMA.COLUMNS
+                 WHERE table_schema = 'pm' AND table_name = %s"""
+        TRN.add(sql, [tbl_name])
+        column_names = [c[0] for c in TRN.execute_fetchindex()]
+        if [tbl_name+'_id', 'name', 'notes'] != column_names:
+            raise LabadminDBError(('Table %s does not have the expected column'
+                                   ' design. It must have exactly the three '
+                                   'columns: "%s_id", "name", "notes"') %
+                                  (tbl_name, tbl_name))
+        return True
+
+
+def _exists(tbl_name, field, value):
+    """ Tests if a value for the given field already exists in the given table.
+
+    Parameters
+    ----------
+    tbl_name: str
+        The DB table name.
+    field: str
+        Name of the table column.
+    value: str
+        Value of the column 'field' in the table 'tbl_name'.
+
+    Returns
+    -------
+    Bool
+        True, iff the table 'tbl_name' exists, has a column 'field' and has no
+        entry with 'value' in column 'field'.
+        False, otherwise.
+
+    Raises
+    ------
+    LabadminDBError
+        a) If the tbl_name does not exist in the DB.
+        b) If the table does not follow the expected design, which is that the
+           table must have exactly the three columns:
+           [tbl_name_id, name, notes]
+    """
+    with TRN:
+        # check that tbl_name exists in DB
+        sql = """SELECT DISTINCT table_name from INFORMATION_SCHEMA.COLUMNS
+                 WHERE table_schema = 'pm' AND table_name = %s"""
+        TRN.add(sql, [tbl_name])
+        if len(TRN.execute_fetchindex()) != 1:
+            raise LabadminDBError('Table %s does not exist in data base.' %
+                                  tbl_name)
+
+        # check that table has the column field
+        sql = """SELECT column_name from INFORMATION_SCHEMA.COLUMNS
+                 WHERE table_schema = 'pm' AND table_name = %s"""
+        TRN.add(sql, [tbl_name])
+        if field not in [c[0] for c in TRN.execute_fetchindex()]:
+            raise LabadminDBError(('Table %s does not have the given column: '
+                                   '%s.') % (tbl_name, field))
+
+        # check if an object with the same value already exists
+        sql = """SELECT """+field+""" FROM pm."""+tbl_name+"""
```

Comment:
Flake8 will complain about this, one option is:
"SELECT {0} FROM pm.{1} WHERE {0} = %s".format(field, tbl_name)